capstone-app/src/app/modules/photos/components/photos-section/photos-section.component.spec.ts

```diff
@@ -1,25 +1,67 @@
+import { PhotosModule } from './../../photos.module';
+import { ImageContainerComponent } from './../image-container/image-container.component';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { MatInputHarness } from '@angular/material/input/testing';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import MockTestImages from 'testing/mock-image-response.json';
+import { of } from 'rxjs';
 import { PhotosSectionComponent } from './photos-section.component';
+import { PhotoFetcher } from '../../services/photo-fetcher.service';
+import { By } from '@angular/platform-browser';
 
 describe('PhotosSectionComponent', () => {
   let component: PhotosSectionComponent;
   let fixture: ComponentFixture<PhotosSectionComponent>;
+  let loader: HarnessLoader;
+  let getPhotosSpy: jasmine.Spy;
 
   beforeEach(async(() => {
+    const photosService = jasmine.createSpyObj('PhotoFetcher', ['getPhotos']);
+    getPhotosSpy = photosService.getPhotos.and.returnValue(of(MockTestImages));
+
+
     TestBed.configureTestingModule({
-      declarations: [ PhotosSectionComponent ]
-    })
-    .compileComponents();
+      declarations: [PhotosSectionComponent, ImageContainerComponent],
+      imports: [PhotosModule],
+      providers: [ { provide: PhotoFetcher, useValue: photosService},
+      ]
+    }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(PhotosSectionComponent);
     component = fixture.componentInstance;
+    loader = TestbedHarnessEnvironment.loader(fixture);
     fixture.detectChanges();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('should ouput the value of the limit', async () => {
+    const inputHarness = await loader.getHarness(
+      MatInputHarness.with({ selector: '.limit-input' })
+    );
+    await inputHarness.setValue('10');
+    const limit = await inputHarness.getValue();
+    expect(limit).toEqual('10');
+  });
+
+  it('should show error message if not in range', async () => {
+    const inputHarness = await loader.getHarness(
+      MatInputHarness.with({ selector: '.limit-input' })
+    );
+    await inputHarness.setValue('15');
+    fixture.whenStable().then(() => {
```

Comment:
I can't recall, was this fixture.whenStable necessary? If not, let's remove it to reduce complexity.